src/applications/search/containers/SearchApp.jsx

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { connect } from 'react-redux';
+
+import { fetchSearchResults } from '../actions';
+import { formatResponseString } from '../utils';
+import { PAGE_SIZE } from '../constants';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import IconSearch from '@department-of-veterans-affairs/formation/IconSearch';
+
+import SimplePagination from '../components/SimplePagination';
+
+class SearchApp extends React.Component {
+  static propTypes = {
+    search: PropTypes.shape({
+      results: PropTypes.array,
+    }).isRequired,
+    fetchSearchResults: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    let userInputFromAddress = '';
+    let offset;
+
+    if (this.props.router.location.query) {
+      userInputFromAddress = this.props.router.location.query.q;
+      offset = this.props.router.location.query.offset;
+    }
+
+    this.state = {
+      userInput: userInputFromAddress,
+      offset,
+    };
+
+    if (!userInputFromAddress) {
+      window.location.href = '/';
+    }
+  }
+
+  componentDidMount() {
+    // If there's data in userInput, it must have come from the address bar, so we immediately hit the API.
+    const { userInput, offset } = this.state;
+    if (userInput) {
+      this.props.fetchSearchResults(userInput, offset);
+    }
+  }
+
+  handleInputChange = event => {
+    this.setState({
+      userInput: event.target.value,
+    });
+  };
+
+  handleSearch = e => {
+    e.preventDefault();
+    const { userInput, offset } = this.state;
+    this.props.router.push({
+      pathname: '',
+      query: {
+        q: encodeURIComponent(userInput),
+        offset,
+      },
+    });
+    this.props.fetchSearchResults(userInput, offset);
+  };
+
+  /* eslint-disable arrow-body-style */
+  handlePageChange = offset => {
+    return e => {
+      e.preventDefault();
+      e.persist();
+      this.setState({ offset }, () => this.handleSearch(e));
+    };
+  };
+  /* eslint-enable arrow-body-style */
+
+  /* eslint-disable react/no-danger */
+  renderWebResult(result) {
+    return (
+      <li key={result.url} className="result-item">
+        <a
+          className="result-title"
+          href={result.url}
+          dangerouslySetInnerHTML={{
+            __html: formatResponseString(result.title),
+          }}
+        />
+        <p className="result-url">{result.url}</p>
+        <p
+          className="result-desc"
+          dangerouslySetInnerHTML={{
+            __html: formatResponseString(result.snippet),
```

Comment:
Is this for sure what we want to be doing here? Not sure if there are already HTML entities or something in there that would be escaped twice, but if it's just to apply the bolded text as done in the `formatResponseString` function, I'm wondering it we could use `string.replace` to iterate though and return JSX. Using `dangerouslySetInnerHTML` to render results from an outside service sounds scary to me 😨